### PR TITLE
Closes #160 | Add data loader for CC-Aligned Sentence Pairs

### LINKED ISSUE #160: Create dataset loader for CC-Aligned Sentence Pairs
Dataloader name: `cc_aligned_sent/cc_aligned_sent.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?cc_aligned_sent

|     Dataset| cc_aligned_sent |
|-------------|---|
| Description | This dataset contains the sentence pairs extracted from CC-Aligned document pairs using similarity scores of LASER embeddings (minimum similarity 1.04, sorted based on decreasing similarity score). It misses some languages not covered by LASER. |
| Subsets | id_ID, jv_ID, su_ID, th_TH, vi_VN, ms_MY, lo_LA, km_KH, my_MM, cx_PH |
| Languages | ind, jav, sun, tha, vie, zlm, lao, khm, mya, ceb |
| Tasks | Machine Translation |
| License | Unknown (unknown) |
| Homepage | https://www2.statmt.org/cc-aligned/ |
| HF URL | - |
| Paper URL | https://aclanthology.org/W19-5435 |

### PR DESCRIPTION
**Pull request by ljvmiranda921**
Resolves #160

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by sabilmakbar** [comment]
whoops, forgot to do this. Let me review the metadata in a bit

File: seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+            target_languages.append(target_lang)
+        else:
+            # Load examples for all languages at once
+            for lang in _LANGUAGES:
+                url = _URLS[lang]
+                source_lang, target_lang = _get_source_target_langs(url)
+
+                source_files.append(dl_manager.download_and_extract(_URLS[lang]))
+                source_languages.append(source_lang)
+                target_languages.append(target_lang)
```
Comment (sabilmakbar) [inline seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py:182]: This also optional

File: seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+class CCAlignedSentencesDataset(datasets.GeneratorBasedBuilder):
+    """CC Aligned Sentences dataset by Chaudhary et al., (2019)"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    SEACROWD_SCHEMA_NAME = "t2t"
+
+    # Add configurations for loading a dataset per language.
+    dataset_names = sorted([f"{_DATASETNAME}_{lang}" for lang in _LANGUAGES])
```
Comment (sabilmakbar) [inline seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py:88]: Imo, it's better if we use the `_SUBSET` instead of `_LANGUAGES` for config creations

**Review by sabilmakbar — COMMENTED** [review]
Init review comments

File: seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+                version=SEACROWD_VERSION,
+                description=f"{_DATASETNAME} SEACrowd schema (all)",
+                schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+                subset_id=_DATASETNAME,
+            ),
+        ]
+    )
+
+    # Choose first language as default
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
```
[the remark this answers is not in the document]
Comment (ljvmiranda921) [reply seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py]: Updated in the latest commit. This simplified the code a lot! Thanks :) https://github.com/SEACrowd/seacrowd-datahub/pull/170/commits/7b89f0cc5282faa1c7a95303ebe5dcfb652a741f

File: seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py (hunk 1 above)
Comment (ljvmiranda921) [reply seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py]: Updated in the latest commit. This simplified the code a lot! Thanks :) https://github.com/SEACrowd/seacrowd-datahub/commit/7b89f0cc5282faa1c7a95303ebe5dcfb652a741f

File: seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py (hunk 2 above)
Comment (ljvmiranda921) [reply seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py]: Got it! Should be in the same commit as well :)

**Comment by ljvmiranda921** [comment]
Ready for another set of reviews! @sabilmakbar @yongzx 🙇

File: seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    def _split_generators(self, dl_manager: DownloadManager) -> List[datasets.SplitGenerator]:
+        """Return SplitGenerators."""
+        # Define some functions for parsing config and URL names
+        def _split_at_n(text: str, n: int) -> Tuple[str, str]:
+            """Split text on the n-th instance"""
+            return ("_".join(text.split("_")[:n]), "_".join(text.split("_")[n:]))
+
+        # Get URL. For cx_PH, the source and target languages are reversed
+        _, subset = _split_at_n(_split_at_n(self.config.name, 5)[0], 3)
+        (source_lang, target_lang) = (subset, "en_XX") if subset == "cx_PH" else ("en_xx", subset)
```
Comment (sabilmakbar) [inline seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py:132]: tried `en_xx`, didn't work. When it's changed to `en_XX` it works like a charm :)
```suggestion
        (source_lang, target_lang) = (subset, "en_XX") if subset == "cx_PH" else ("en_XX", subset)
```

**Review by sabilmakbar — CHANGES_REQUESTED** [review]
one change on dataloader URL

**Review by sabilmakbar — APPROVED** [review]
all okay and tested. LGTM, thanks LJ! Let's wait for @yongzx's review

**Comment by yongzx** [comment]
Tested as well and everything works! I'll merge this.

**Pull request merged**